### PR TITLE
Zoom

### PR DESCRIPTION
**Pull request by AirOxa-Deepesh**
## Description

created a fully functional zoom integration for elba security which passes all test cases as  mentioned in the issue no 5,6,and 7.

## Checklist:

Before you create this PR, confirm all the requirements listed below by checking the checkboxes like this (`[x]`).

- [ x] I have performed a self-review of my own code.
- [ x] I have commented my code, particularly in hard-to-understand areas.
- [ x] I have added tests to cover the new feature or fixes.

### CODE REVIEW

File: apps/zoom/src/inngest/functions/users/sync-users-page.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+    // retrieve the SaaS organisation token
+
+    const nextPage = await step.run('list-users', async () => {
+      const result = await getUsers(token, page);
+      // format each SaaS users to elba users
+      const users = result.users.map(formatElbaUser);
+
+      // send the batch of users to elba
+      // TODO: will do it later because i need elba working url and api key
```
[the remark this answers is not in the document]
Comment (AirOxa-Deepesh) [reply apps/zoom/src/inngest/functions/users/sync-users-page.ts:74]: i have removed the comments. They were just for my understanding.

File: apps/zoom/src/app/api/inngest/route.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+// export const preferredRegion = env.VERCEL_PREFERRED_REGION;
+// Remove the next line if your integration does not works with edge runtime
+// export const runtime = 'edge';
+export const dynamic = 'force-dynamic';
+
+export const { GET, POST, PUT } = serve({
+  client: inngest,
+  functions: inngestFunctions,
+  // Remove the next line if your integration does not works with edge runtime
+  // streaming: 'allow',
```
[the remark this answers is not in the document]
Comment (AirOxa-Deepesh) [reply apps/zoom/src/app/api/inngest/route.ts:15]: ok.i will re-enabled the edge runtime.

File: apps/zoom/src/database/schema.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+export const Organisation = pgTable('organisation', {
+  id: uuid('id').primaryKey(),
+
+  region: text('region').notNull(),
+  createdAt: timestamp('created_at').defaultNow().notNull(),
+  // the following properties are examples, it can removed / replaced to fit your use-case
+
+  accessToken: text('access_token').notNull(),
+  refreshToken: text('refresh_token').notNull(),
+  expiresIn: timestamp('expire_at').notNull(),
```
[the remark this answers is not in the document]
Comment (AirOxa-Deepesh) [reply apps/zoom/src/database/schema.ts:13]: Noted. i will add the comment.

File: apps/zoom/src/connectors/users.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+  if (page !== null) {
+    zoomUrl.searchParams.append('next_page_token', page);
+  }
+
+  const response = await fetch(zoomUrl.toString(), {
+    headers: { Authorization: `Bearer ${token}` },
+  });
+
+  if (!response.ok || response.status === 401) {
```
Comment (elbalexandre) [inline apps/zoom/src/connectors/users.ts:43]: if response.status > 2XX then response.ok is false anyway

```suggestion
  if (!response.ok) {
```

File: apps/zoom/src/inngest/functions/users/sync-users-page.test.ts [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+    // check that the function continue the pagination process
+    expect(step.sendEvent).toBeCalledTimes(1);
+    /* eslint-disable -- no type here */
+    expect(step.sendEvent).toBeCalledWith('sync-users-page', {
+      name: 'zoom/users.page_sync.requested',
+      data: {
+        organisationId: organisation.id,
+        isFirstSync: false,
+        syncStartedAt: expect.any(Number),
```
[the remark this answers is not in the document]
Comment (elbalexandre) [reply apps/zoom/src/inngest/functions/users/sync-users-page.test.ts:82]: So you can expect this value here `now.getTime()`

**Review by elbalexandre — APPROVED** [review]

**Pull request merged**